packages/web3-eth-accounts/src/account.ts

```diff
@@ -1,22 +1,166 @@
+	if (privateKey.length !== 66) {
+		throw new Error('Private key must be 32 bytes long');
+	}
+
+	const hash = hashMessage(data);
+
+	const signObj = ecdsaSign(
+		Buffer.from(hash.substring(2), 'hex'),
+		Buffer.from(privateKey.substring(2), 'hex'),
+	);
+
+	const r = Buffer.from(signObj.signature.slice(0, 32));
+	const s = Buffer.from(signObj.signature.slice(32, 64));
+	const v = signObj.recid + 27;
+
+	return {
+		message: data,
+		messageHash: hash,
+		v: `0x${v.toString(16)}`,
+		r: `0x${r.toString('hex')}`,
+		s: `0x${s.toString('hex')}`,
+		signature: `0x${Buffer.from(signObj.signature).toString('hex')}${v.toString(16)}`,
+	};
+};
+
+/**
+ *  Signs an Ethereum transaction with a given private key.
+ */
+export const signTransaction = (
+	transaction: TxData | AccessListEIP2930TxData | FeeMarketEIP1559TxData,
+	privateKey: string,
+): signTransactionResult => {
+	//	TO DO : Send calls to web3.transaction package for :
+	//		Transaction Validation checks
+
+	const tx = TransactionFactory.fromTxData(transaction);
+	const signedTx = tx.sign(Buffer.from(privateKey.substring(2), 'hex'));
+	if (signedTx.v === undefined || signedTx.r === undefined || signedTx.s === undefined)
+		throw new Error('Signer Error');
+
+	const validationErrors = signedTx.validate(true);
+
+	if (validationErrors.length > 0) {
+		let errorString = 'Signer Error: ';
+		for (const validationError of validationErrors) {
+			errorString += `${errorString} ${validationError}.`;
+		}
+		throw new Error(errorString);
+	}
+
+	const rlpEncoded = signedTx.serialize().toString('hex');
+	const rawTx = `0x${rlpEncoded}`;
+	const txHash = keccak256(Buffer.from(rawTx, 'hex'));
+
+	return {
+		messageHash: `0x${Buffer.from(signedTx.getMessageToSign(true)).toString('hex')}`,
+		v: `0x${signedTx.v.toString('hex')}`,
+		r: `0x${signedTx.r.toString('hex')}`,
+		s: `0x${signedTx.s.toString('hex')}`,
+		rawTransaction: rawTx,
+		transactionHash: `0x${Buffer.from(txHash).toString('hex')}`,
+	};
+};
+
+/**
+ * Recovers the Ethereum address which was used to sign the given RLP encoded transaction.
+ */
+export const recoverTransaction = (rawTransaction: string): Address => {
+	if (rawTransaction === undefined) throw new Error('Invalid rawTransaction');
+
+	const tx = TransactionFactory.fromSerializedData(Buffer.from(rawTransaction.slice(2), 'hex'));
+
+	return toChecksumAddress(tx.getSenderAddress().toString());
+};
+
+/**
+ * Recovers the Ethereum address which was used to sign the given data
+ */
+export const recover = (
+	data: string | signatureObject,
+	signature?: string,
+	hashed?: boolean,
+): Address => {
+	if (typeof data === 'object') {
+		const signatureStr = `${data.r}${data.s.slice(2)}${data.v.slice(2)}`;
+		return recover(data.messageHash, signatureStr, true);
+	}
+
+	if (signature === undefined) throw new Error('signature string undefined');
+
+	const hashedMessage = hashed ? data : hashMessage(data);
+
+	const v = signature.substring(130); // 0x + r + s + v
```

Comment:
It's preferable to avoid magic numbers in the code. Try creating constants for it then use it here. 

Does it really pass the linting? We had following rule enabled for `src`. 

https://eslint.org/docs/rules/no-magic-numbers

Reply:
made this change. It was passing as can be seen in PR CI / CD actions.